### PR TITLE
feat(create-workspace): Add a script to quickly create a new workspace from the template

### PR DESCRIPTION
**Pull request by gribnoysup**
I noticed that our CONTRIBUTING section about creating a new package got kinda long and consisting of too many steps to do manually, there was also some undocumented things about the npm scripts that caused some CI failures recently. I started with just updating the docs, but it seems to me that having some tool that will ask you a few questions and bootstrap everything itself might be a better approach here. This is how this script looks in action:

Kapture 2021-08-23 at 11 55 59

I also moved all the config files into their own `configs` folder so it's a bit easier to find them and they are not in the way when looking for packages.

[Rendered CONTRIBUTING.md](https://github.com/mongodb-js/compass/blob/958074fc440c65fe2efce06082116dc0149a5b83/CONTRIBUTING.md) for convenience.

### CODE REVIEW

File: packages/data-service/package.json [hunk 1]
```diff
@@ -43,15 +43,12 @@
     "bootstrap": "npm run compile",
     "prepublishOnly": "npm run compile",
     "pretest": "npm run compile && mongodb-runner start --port=27018",
-    "test": "cross-env TS_NODE_FILES=true nyc mocha --colors --timeout 15000 -r ts-node/register src/**/*.spec.*",
+    "test": "cross-env TS_NODE_FILES=true nyc mocha",
```
Comment (mcasimir) [inline packages/data-service/package.json:46]: was fighting with this a lot, can we have a different task for covered tests (`test-cov`)? And a `test-watch` with `--watch`, maybe only me, but personally I use that a ton and run tests each 3 seconds, i would love to have a way for that to be really fast.

File: scripts/create-workspace.js [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+      prepublishOnly: 'npm run compile',
+      compile:
+        'tsc -p tsconfig.json && gen-esm-wrapper . ./dist/.esm-wrapper.mjs',
+      eslint: 'eslint',
+      prettier: 'prettier',
+      lint: 'npm run eslint . && npm run prettier -- --check .',
+      depcheck: 'depcheck',
+      check: 'npm run lint && npm run depcheck',
+      'check-ci': 'npm run check',
+      test: 'cross-env TS_NODE_FILES=true nyc mocha',
```
Comment (mcasimir) [inline scripts/create-workspace.js:173]: was fighting with this a lot, can we have a different task for covered tests (`test-cov`)? And a `test-watch` with `--watch`, maybe only me, but personally I use that a ton and run tests each 3 seconds, i would love to have a way for that to be really fast.

Would you be ok with including the changes i've done here: (also fixes nyc reporting coverage in test files)

https://github.com/mongodb-js/compass/pull/2412/files#diff-a557af5d61219f0b3f0b1fc6810efa3a9e0889fc54005cd248908a50cdf1d612L40

File: scripts/create-workspace.js [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+        canceled = true;
+      }
+    }
+  );
+
+  if (canceled) {
+    return;
+  }
+
+  console.log();
```
Comment (mcasimir) [inline scripts/create-workspace.js:134]: is this a newline?

File: configs/mocha-config-compass/react.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const path = require('path');
+const base = require('./');
+
+module.exports = {
+  ...base,
+  require: base.require.concat([
+    'jsdom-global/register',
+    path.resolve(__dirname, 'chai-dom-register.js'),
```
Comment (mcasimir) [inline configs/mocha-config-compass/react.js:8]: should we add also `sinon` and `sinon-chai` somewhere?

**Review by mcasimir — APPROVED** [review]
Amazing! Great to have these config packages in its own folder, and the mocha config is super neat.

File: scripts/create-workspace.js (hunk 3 above)
Comment (gribnoysup) [reply scripts/create-workspace.js:134]: Yep

File: configs/mocha-config-compass/react.js (hunk 4 above)
Comment (gribnoysup) [reply configs/mocha-config-compass/react.js:8]: Yep, good idea, seems like we use it a lot in tests suites (I somehow never noticed before)

File: scripts/create-workspace.js (hunk 2 above)
Comment (gribnoysup) [reply scripts/create-workspace.js:173]: Totally, let's include this stuff in the template! Then following your suggestion we will scripts like this:

- `test` -> `mocha`
- `test-cov` -> `cross-env TS_NODE_FILES=true nyc -x "**/*.spec.*" npm run test`
- `test-watch` -> `npm run test -- --watch`
- `test-ci` -> `npm run test-cov`

Makes sense?

Comment (mcasimir) [reply scripts/create-workspace.js:173]: Yep!

Comment (gribnoysup) [reply scripts/create-workspace.js]: Done in 34089b9

File: configs/mocha-config-compass/react.js (hunk 4 above)
Comment (gribnoysup) [reply configs/mocha-config-compass/react.js:8]: Done in a74c870

**Pull request merged**